// @azure/arm-appcomplianceautomation/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link Origin} that the service accepts. */
export var KnownOrigin;
(function (KnownOrigin) {
    /** User */
    KnownOrigin["User"] = "user";
    /** System */
    KnownOrigin["System"] = "system";
    /** UserSystem */
    KnownOrigin["UserSystem"] = "user,system";
})(KnownOrigin || (KnownOrigin = {}));
/** Known values of {@link ActionType} that the service accepts. */
export var KnownActionType;
(function (KnownActionType) {
    /** Internal */
    KnownActionType["Internal"] = "Internal";
})(KnownActionType || (KnownActionType = {}));
/** Known values of {@link ReportStatus} that the service accepts. */
export var KnownReportStatus;
(function (KnownReportStatus) {
    /** Active */
    KnownReportStatus["Active"] = "Active";
    /** Failed */
    KnownReportStatus["Failed"] = "Failed";
    /** Disabled */
    KnownReportStatus["Disabled"] = "Disabled";
})(KnownReportStatus || (KnownReportStatus = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** Succeeded */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** Failed */
    KnownProvisioningState["Failed"] = "Failed";
    /** Canceled */
    KnownProvisioningState["Canceled"] = "Canceled";
    /** Creating */
    KnownProvisioningState["Creating"] = "Creating";
    /** Deleting */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** Updating */
    KnownProvisioningState["Updating"] = "Updating";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link CreatedByType} that the service accepts. */
export var KnownCreatedByType;
(function (KnownCreatedByType) {
    /** User */
    KnownCreatedByType["User"] = "User";
    /** Application */
    KnownCreatedByType["Application"] = "Application";
    /** ManagedIdentity */
    KnownCreatedByType["ManagedIdentity"] = "ManagedIdentity";
    /** Key */
    KnownCreatedByType["Key"] = "Key";
})(KnownCreatedByType || (KnownCreatedByType = {}));
/** Known values of {@link CategoryType} that the service accepts. */
export var KnownCategoryType;
(function (KnownCategoryType) {
    /** FullyAutomated */
    KnownCategoryType["FullyAutomated"] = "FullyAutomated";
    /** PartiallyAutomated */
    KnownCategoryType["PartiallyAutomated"] = "PartiallyAutomated";
    /** Manual */
    KnownCategoryType["Manual"] = "Manual";
})(KnownCategoryType || (KnownCategoryType = {}));
/** Known values of {@link CategoryStatus} that the service accepts. */
export var KnownCategoryStatus;
(function (KnownCategoryStatus) {
    /** Healthy */
    KnownCategoryStatus["Healthy"] = "Healthy";
    /** Unhealthy */
    KnownCategoryStatus["Unhealthy"] = "Unhealthy";
})(KnownCategoryStatus || (KnownCategoryStatus = {}));
/** Known values of {@link ControlFamilyType} that the service accepts. */
export var KnownControlFamilyType;
(function (KnownControlFamilyType) {
    /** FullyAutomated */
    KnownControlFamilyType["FullyAutomated"] = "FullyAutomated";
    /** PartiallyAutomated */
    KnownControlFamilyType["PartiallyAutomated"] = "PartiallyAutomated";
    /** Manual */
    KnownControlFamilyType["Manual"] = "Manual";
})(KnownControlFamilyType || (KnownControlFamilyType = {}));
/** Known values of {@link ControlFamilyStatus} that the service accepts. */
export var KnownControlFamilyStatus;
(function (KnownControlFamilyStatus) {
    /** Healthy */
    KnownControlFamilyStatus["Healthy"] = "Healthy";
    /** Unhealthy */
    KnownControlFamilyStatus["Unhealthy"] = "Unhealthy";
})(KnownControlFamilyStatus || (KnownControlFamilyStatus = {}));
/** Known values of {@link ControlType} that the service accepts. */
export var KnownControlType;
(function (KnownControlType) {
    /** FullyAutomated */
    KnownControlType["FullyAutomated"] = "FullyAutomated";
    /** PartiallyAutomated */
    KnownControlType["PartiallyAutomated"] = "PartiallyAutomated";
    /** Manual */
    KnownControlType["Manual"] = "Manual";
})(KnownControlType || (KnownControlType = {}));
/** Known values of {@link ControlStatus} that the service accepts. */
export var KnownControlStatus;
(function (KnownControlStatus) {
    /** Passed */
    KnownControlStatus["Passed"] = "Passed";
    /** Failed */
    KnownControlStatus["Failed"] = "Failed";
    /** NotApplicable */
    KnownControlStatus["NotApplicable"] = "NotApplicable";
})(KnownControlStatus || (KnownControlStatus = {}));
/** Known values of {@link AssessmentSeverity} that the service accepts. */
export var KnownAssessmentSeverity;
(function (KnownAssessmentSeverity) {
    /** High */
    KnownAssessmentSeverity["High"] = "High";
    /** Medium */
    KnownAssessmentSeverity["Medium"] = "Medium";
    /** Low */
    KnownAssessmentSeverity["Low"] = "Low";
})(KnownAssessmentSeverity || (KnownAssessmentSeverity = {}));
/** Known values of {@link IsPass} that the service accepts. */
export var KnownIsPass;
(function (KnownIsPass) {
    /** True */
    KnownIsPass["True"] = "True";
    /** False */
    KnownIsPass["False"] = "False";
})(KnownIsPass || (KnownIsPass = {}));
/** Known values of {@link ResourceStatus} that the service accepts. */
export var KnownResourceStatus;
(function (KnownResourceStatus) {
    /** Healthy */
    KnownResourceStatus["Healthy"] = "Healthy";
    /** Unhealthy */
    KnownResourceStatus["Unhealthy"] = "Unhealthy";
    /** NotApplicable */
    KnownResourceStatus["NotApplicable"] = "NotApplicable";
})(KnownResourceStatus || (KnownResourceStatus = {}));
/** Known values of {@link DownloadType} that the service accepts. */
export var KnownDownloadType;
(function (KnownDownloadType) {
    /** ComplianceReport */
    KnownDownloadType["ComplianceReport"] = "ComplianceReport";
    /** CompliancePdfReport */
    KnownDownloadType["CompliancePdfReport"] = "CompliancePdfReport";
    /** ComplianceDetailedPdfReport */
    KnownDownloadType["ComplianceDetailedPdfReport"] = "ComplianceDetailedPdfReport";
    /** ResourceList */
    KnownDownloadType["ResourceList"] = "ResourceList";
})(KnownDownloadType || (KnownDownloadType = {}));
/** Known values of {@link ComplianceState} that the service accepts. */
export var KnownComplianceState;
(function (KnownComplianceState) {
    /** Healthy */
    KnownComplianceState["Healthy"] = "Healthy";
    /** Unhealthy */
    KnownComplianceState["Unhealthy"] = "Unhealthy";
})(KnownComplianceState || (KnownComplianceState = {}));
