// @azure/arm-appcomplianceautomation/dist-esm/src/operations/snapshot.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __awaiter } from "tslib";
import * as coreClient from "@azure/core-client";
import * as Mappers from "../models/mappers";
import * as Parameters from "../models/parameters";
import { LroEngine } from "@azure/core-lro";
import { LroImpl } from "../lroImpl";
/** Class containing Snapshot operations. */
export class SnapshotImpl {
    /**
     * Initialize a new instance of the class Snapshot class.
     * @param client Reference to the service client
     */
    constructor(client) {
        this.client = client;
    }
    /**
     * Get the AppComplianceAutomation snapshot and its properties.
     * @param reportName Report Name.
     * @param snapshotName Snapshot Name.
     * @param options The options parameters.
     */
    get(reportName, snapshotName, options) {
        return this.client.sendOperationRequest({ reportName, snapshotName, options }, getOperationSpec);
    }
    /**
     * Download compliance needs from snapshot, like: Compliance Report, Resource List.
     * @param reportName Report Name.
     * @param snapshotName Snapshot Name.
     * @param parameters Parameters for the query operation
     * @param options The options parameters.
     */
    beginDownload(reportName, snapshotName, parameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const directSendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                return this.client.sendOperationRequest(args, spec);
            });
            const sendOperation = (args, spec) => __awaiter(this, void 0, void 0, function* () {
                var _a;
                let currentRawResponse = undefined;
                const providedCallback = (_a = args.options) === null || _a === void 0 ? void 0 : _a.onResponse;
                const callback = (rawResponse, flatResponse) => {
                    currentRawResponse = rawResponse;
                    providedCallback === null || providedCallback === void 0 ? void 0 : providedCallback(rawResponse, flatResponse);
                };
                const updatedArgs = Object.assign(Object.assign({}, args), { options: Object.assign(Object.assign({}, args.options), { onResponse: callback }) });
                const flatResponse = yield directSendOperation(updatedArgs, spec);
                return {
                    flatResponse,
                    rawResponse: {
                        statusCode: currentRawResponse.status,
                        body: currentRawResponse.parsedBody,
                        headers: currentRawResponse.headers.toJSON()
                    }
                };
            });
            const lro = new LroImpl(sendOperation, { reportName, snapshotName, parameters, options }, downloadOperationSpec);
            const poller = new LroEngine(lro, {
                resumeFrom: options === null || options === void 0 ? void 0 : options.resumeFrom,
                intervalInMs: options === null || options === void 0 ? void 0 : options.updateIntervalInMs,
                lroResourceLocationConfig: "azure-async-operation"
            });
            yield poller.poll();
            return poller;
        });
    }
    /**
     * Download compliance needs from snapshot, like: Compliance Report, Resource List.
     * @param reportName Report Name.
     * @param snapshotName Snapshot Name.
     * @param parameters Parameters for the query operation
     * @param options The options parameters.
     */
    beginDownloadAndWait(reportName, snapshotName, parameters, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const poller = yield this.beginDownload(reportName, snapshotName, parameters, options);
            return poller.pollUntilDone();
        });
    }
}
// Operation Specifications
const serializer = coreClient.createSerializer(Mappers, /* isXml */ false);
const getOperationSpec = {
    path: "/providers/Microsoft.AppComplianceAutomation/reports/{reportName}/snapshots/{snapshotName}",
    httpMethod: "GET",
    responses: {
        200: {
            bodyMapper: Mappers.SnapshotResource
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.reportName,
        Parameters.snapshotName
    ],
    headerParameters: [Parameters.accept],
    serializer
};
const downloadOperationSpec = {
    path: "/providers/Microsoft.AppComplianceAutomation/reports/{reportName}/snapshots/{snapshotName}/download",
    httpMethod: "POST",
    responses: {
        200: {
            bodyMapper: Mappers.DownloadResponse
        },
        201: {
            bodyMapper: Mappers.DownloadResponse
        },
        202: {
            bodyMapper: Mappers.DownloadResponse
        },
        204: {
            bodyMapper: Mappers.DownloadResponse
        },
        default: {
            bodyMapper: Mappers.ErrorResponse
        }
    },
    requestBody: Parameters.parameters2,
    queryParameters: [Parameters.apiVersion],
    urlParameters: [
        Parameters.$host,
        Parameters.reportName,
        Parameters.snapshotName
    ],
    headerParameters: [Parameters.accept, Parameters.contentType],
    mediaType: "json",
    serializer
};
